README.md

```diff
@@ -1 +1,108 @@
-initial commit
+# Custom AWS Identity Provider (for AWS Connect)
+
+Dynamically generate SAML Response XML documents from a custom Identity Provider
+without running a server. The primary goal is to create SAML Responses which can
+be used for logging in to AWS Connect.
+
+## Background
+
+This code was adopted from the [Keycloak](https://www.keycloak.org/) project
+([source code](https://github.com/keycloak/keycloak)) in order to host a custom
+SAML-based Identity Provider (IdP) without having to run a server or rely on
+some external service. Other tools were investigated in JavaScript 
+([@node-saml/node-saml](https://github.com/node-saml/node-saml/),
+[xml-crypto](https://github.com/node-saml/xml-crypto/),
+[saml-idp](https://github.com/mcguinness/saml-idp), and
+[passport-saml](http://www.passportjs.org/packages/passport-saml/)), but they
+all seemed to be bad fits (node-saml explicitly is intended to be used on the
+Service Provider (SP) side, I was unable to generate a valid signature with
+xml-crypto, saml-idp had insecure dependencies, and passport-saml depends on
+[ExpressJS](https://expressjs.com/) which we aren't using).
```

Comment:
just for my own interest, can you expand on why passport-saml depending on ExpressJS is an issue? I thought we use Express quite often

Reply:
I should actually probably revisit passport, it seems like it may do what I want. For now, I have updated the readme to mention the same.